packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -391,7 +412,95 @@ abstract class _DeeplinkApiHandler {
     if (missingRequiredParams != null) return missingRequiredParams;
 
     final rootPath = queryParams[DeeplinkApi.deeplinkRootPathPropertyName]!;
-    final result = await deeplinkManager.getBuildVariants(rootPath: rootPath);
+    final result =
+        await deeplinkManager.getAndroidBuildVariants(rootPath: rootPath);
+    final error = result[DeeplinkManager.kErrorField] as String?;
+    if (error != null) {
+      api.serverError(error);
+    }
+    return api.getCompleted(
+      result[DeeplinkManager.kOutputJsonField]! as String,
+    );
+  }
+
+  static Future<shelf.Response> handleAndroidAppLinkSettings(
+    ServerApi api,
+    Map<String, String> queryParams,
+    DeeplinkManager deeplinkManager,
+  ) async {
+    final missingRequiredParams = ServerApi._checkRequiredParameters(
+      [
+        DeeplinkApi.deeplinkRootPathPropertyName,
+        DeeplinkApi.androidBuildVariantPropertyName,
+      ],
+      queryParams: queryParams,
+      api: api,
+      requestName: DeeplinkApi.androidBuildVariants,
+    );
+    if (missingRequiredParams != null) return missingRequiredParams;
+
+    final rootPath = queryParams[DeeplinkApi.deeplinkRootPathPropertyName]!;
+    final buildVariant =
+        queryParams[DeeplinkApi.androidBuildVariantPropertyName]!;
+    final result = await deeplinkManager.getAndroidAppLinkSettings(
+      rootPath: rootPath,
+      buildVariant: buildVariant,
+    );
+    final error = result[DeeplinkManager.kErrorField] as String?;
+    if (error != null) {
+      api.serverError(error);
+    }
+    return api.getCompleted(
+      result[DeeplinkManager.kOutputJsonField]! as String,
+    );
+  }
+
+  static Future<shelf.Response> handleIosBuildOptions(
+    ServerApi api,
+    Map<String, String> queryParams,
+    DeeplinkManager deeplinkManager,
+  ) async {
+    final missingRequiredParams = ServerApi._checkRequiredParameters(
+      [DeeplinkApi.deeplinkRootPathPropertyName],
+      queryParams: queryParams,
+      api: api,
+      requestName: DeeplinkApi.iosBuildOptions,
+    );
+    if (missingRequiredParams != null) return missingRequiredParams;
+
+    final rootPath = queryParams[DeeplinkApi.deeplinkRootPathPropertyName]!;
+    final result = await deeplinkManager.getIosBuildOptions(rootPath: rootPath);
+    final error = result[DeeplinkManager.kErrorField] as String?;
+    if (error != null) {
+      api.serverError(error);
+    }
+    return api.getCompleted(
+      result[DeeplinkManager.kOutputJsonField]! as String,
+    );
```

Comment:
this block is duplicated a few times. Can we use a helper `resultOutputOrError(result)`?